### ISSUE TITLE
Split Repo: Server

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/isaac#57]
Since the server and clients share no code with the isaac library (not even headers) we should split them in separate repos.

This will ease packing, e.g., [spack](https://github.com/LLNL/spack) recipes for library & server.

- add "protocol_version" attribute to json so `server` and `client` can check & abort on protocol mismatch for further development with breaking in-JSON-protocol changes

**Comment by theZiz** [comment]
I guess this is a good idea, but in which direction? A new server repository or a new library repository? I guess the first idea makes more sense as PIConGPU and the documentation of it does not needed to get touched for this.

But nevertheless: What is the problem creating two recipes for spack? They both use the same repository, but could also use different CMake-variables. So maybe just improving the CMakeLists.txt may make sense, too?

**Comment by ax3l** [comment]
my suggestion similar to popular naming schemes:
- keep the library in `isaac`
- move the clients to `isaac-clients` or `isaac-html`
- move the server to `isaac-server`

and don't forget about he `protocol_version` stringy.

> But nevertheless: What is the problem creating two recipes for spack? They both use the same repository, but could also use different CMake-variables. So maybe just improving the CMakeLists.txt may make sense, too?

Yes, allowing to build & install the server and library cleanly and separately would also help to solve that issue. I would still introduce a `protocol_version` since we already use independent server and library versions of ISAAC in production.

**Comment by theZiz** [comment]
Althought this should have been in another issue, I implemented the protocol version check: https://github.com/ComputationalRadiationPhysics/isaac/commit/6ab1b3d472fe9623f22466d7a2e47e05b8f547e2
Furthermore I made it optional to not build/install the server or the library. I am still not sure, whether it is a good idea to split the repositories. The solution should work with sparck for now: https://github.com/ComputationalRadiationPhysics/isaac/commit/62156d3ad53b5d316fe310f35eda2d4330c381bc

**Comment by ax3l** [comment]
awesome, thanks!

yeah, the split of at least the `CMakeLists.txt` files would be useful, since flags such as `-DISAAC_GST=OFF` make no sense for a library-only or client-only install.

**Comment by theZiz** [comment]
Yes. But if you make `cmake -DISAAC_BUILD_SERVER=OFF ..` it doesn't show ISAAC_GST to you (at least in ccmake). :shipit:

**Comment by ax3l** [comment]
but `INSTALL.md` does and it confuses people trying to use specific aspect of ISAAC.

We should split the `CMakeLists.txt`, client and server at least in the repo so they look like, build and install individual projects depending on one another

**Comment by theZiz** [comment]
Okay, sounds like a good compromise.

**Comment by theZiz** [comment]
Done: https://github.com/ComputationalRadiationPhysics/isaac/commit/1801b979de2bd5b3be4475eb6a05d3672bdcf8b2
Works fine on my machine, but I can't test today on the cluster for obvious reasons. ;)

**Comment by ax3l** [comment]
thanks, added some inline comments in the commit above regarding the updates in `INSTALL.md`

**Comment by theZiz** [comment]
Thanks for you comments. Fixed them: https://github.com/ComputationalRadiationPhysics/isaac/commit/dcf4a0f35c1b165dcc9bb46616be4d1a5c298259
